### PR TITLE
vermin: init a 1.6.0

### PR DESCRIPTION
**Pull request by fidgetingbits**
Add vermin tool. It's a tool for checking python version compatibility.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ve/vermin/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  ];
+
+  checkPhase = ''
+    python runtests.py
+  '';
+
+  meta = {
+    homepage = "https://github.com/netromdk/vermin";
+    changelog = "https://github.com/netromdk/vermin/releases/tag/v${version}";
+    description = "Concurrently detect the minimum Python versions needed to run code.";
```
[the remark this answers is not in the document]
Comment (fidgetingbits) [reply pkgs/by-name/ve/vermin/package.nix]: Fixed

File: pkgs/by-name/ve/vermin/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3,
+}:
+python3.pkgs.buildPythonApplication rec {
```
[the remark this answers is not in the document]
Comment (fidgetingbits) [reply pkgs/by-name/ve/vermin/package.nix]: Fixed

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/ve/vermin/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  dependencies = with python3Packages; [
+    setuptools
+  ];
+
+  checkPhase = ''
+    python runtests.py
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/vermin/package.nix:27]:
```suggestion
    runHook preCheck
    python runtests.py
    runHook postCheck
```

File: pkgs/by-name/ve/vermin/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  '';
+
+  meta = {
+    homepage = "https://github.com/netromdk/vermin";
+    changelog = "https://github.com/netromdk/vermin/releases/tag/v${version}";
+    description = "Concurrently detect the minimum Python versions needed to run code";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.fidgetingbits ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/vermin/package.nix:35]: missing `mainProgram`

File: pkgs/by-name/ve/vermin/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  python3Packages,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "vermin";
+  version = "1.6.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "netromdk";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/vermin/package.nix:13]:
```suggestion
    repo = "vermin";
```

**Comment by fidgetingbits** [comment]
Thank you @Aleksanaa , all issues should be fixed.

**Review by Aleksanaa — APPROVED** [review]
tested on aarch64-darwin

**Pull request merged**